### PR TITLE
Added support for text search endpoint

### LINKED ISSUE #292: Add text search support to the driver
Update the driver to support the new text search functionality.

### PR DESCRIPTION
**Pull request by r-marques**
resolves #292 

- Added support for the assets endpoint
- Added tests
- Added documentation

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -566,3 +568,62 @@ The ``fulfilled_transfer_tx`` dictionary should have two outputs, one with
...
+            'id': 'e40f4b6ac70b9c1b3b237ec13f4174384fd4d54d36dfde25520171577c49caa4'
+        },
+        {
+            'data': {'msg': 'Hello BigchainDB 3!'},
+            'id': '748f6c30daaf771c9020d84db9ad8ac4d1f7c8de7013db55e16f10ba090f7013'
+        }
+    ]
+
+This call returns all the assets that match the string ``bigchaindb``, sorted
+by text score, as well as the asset ``id``. This is the same ``id`` of the
```
Comment (TimDaub) [inline docs/usage.rst]: Should we link to the mongodb documentation here?

https://docs.mongodb.com/manual/reference/operator/query/text/#text-operator-text-score

File: docs/usage.rst [hunk 2]
```diff
@@ -566,3 +568,64 @@ The ``fulfilled_transfer_tx`` dictionary should have two outputs, one with
...
+        }
+    ]
+
+This call returns all the assets that match the string ``bigchaindb``, sorted
+by `text score
+<https://docs.mongodb.com/manual/reference/operator/query/text/#text-operator-text-score>`_,
+as well as the asset ``id``. This is the same ``id`` of the transaction that
+created the asset.
+
+Its also possible to limit the amount of returned results using the ``limit``
```
Comment (sbellem) [inline docs/usage.rst]: Its --> It's

File: docs/usage.rst [hunk 3]
```diff
@@ -566,3 +568,64 @@ The ``fulfilled_transfer_tx`` dictionary should have two outputs, one with
...
+
+.. ipython::
+
+    In [0]: assets = [
+       ...:    {'data': {'msg': 'Hello BigchainDB 1!'}},
+       ...:    {'data': {'msg': 'Hello BigchainDB 2!'}},
+       ...:    {'data': {'msg': 'Hello BigchainDB 3!'}}
+       ...: ]
+
+Lets perform a text search for all assets that contain the word ``bigchaindb``:
```
Comment (sbellem) [inline docs/usage.rst]: Lets --> Let's

File: docs/usage.rst [hunk 4]
```diff
@@ -566,3 +568,64 @@ The ``fulfilled_transfer_tx`` dictionary should have two outputs, one with
...
     In [0]: fulfilled_transfer_tx
+
+Querying for Assets
+-------------------
+
+BigchainDB allows you to query for assets using simple text search. This search
+is applied to all the strings inside the asset payload and returns all the
+assets that match a give text search string.
+
+Lets assume that we :ref:`created <asset-creation>` 3 assets that look like
```
Comment (sbellem) [inline docs/usage.rst]: Lets --> Let's

File: docs/usage.rst [hunk 5]
```diff
@@ -566,3 +568,64 @@ The ``fulfilled_transfer_tx`` dictionary should have two outputs, one with
 .. ipython::
 
     In [0]: fulfilled_transfer_tx
+
+Querying for Assets
+-------------------
+
+BigchainDB allows you to query for assets using simple text search. This search
+is applied to all the strings inside the asset payload and returns all the
+assets that match a give text search string.
```
Comment (sbellem) [inline docs/usage.rst]: give --> given

**Comment by r-marques** [comment]
Updated the test environment to skip the assets test if the backend is not mongodb.
The tests should pass after #297

**Pull request merged**